clip/eval.py

```diff
@@ -0,0 +1,55 @@
+# Mirror of the Linear Probe Evaluation Script
+# from the official CLIP Repository.
+import mlx.core as mx
+import numpy as np
+from image_processor import CLIPImageProcessor
+from mlx.data.datasets import load_cifar10
+from model import CLIPModel
+from PIL import Image
+from sklearn.linear_model import LogisticRegression
+
+
+def get_cifar10(batch_size, root=None):
+    tr = load_cifar10(root=root)
+    tr_iter = tr.to_stream().batch(batch_size)
+
+    test = load_cifar10(root=root, train=False)
+    test_iter = test.to_stream().batch(batch_size)
```

Comment:
Since it is a very small in memory dataset the stream is unnecessary here. I 'd keep it a buffer so that it is nicer and we have a `len` as well.

Namely,

```python
train_iter = load_cifar10(root=root).batch(batch_size)
```

Reply:
Fixed in https://github.com/ml-explore/mlx-examples/pull/960/commits/dee703eacf2c8ae1a5c917426b2d38aa1db77ddc